tests/DCPS/Thrasher/ParticipantTask.cpp

```diff
@@ -168,8 +180,8 @@ ParticipantTask::svc()
     // pathways related to publication; we should be especially dull
     // and write only one sample at a time per writer.
 
-    ProgressIndicator progress("(%P|%t)       PUBLISHER %d%% (%d samples sent)\n",
-                               samples_per_thread_);
+    const std::string fmt(pfx + "  %d%% (%d samples sent)\n");
+    ProgressIndicator progress(fmt.c_str(), samples_per_thread_);
```

Comment:
Make the progress indicator take a thread id to avoid the dynamic format string.

Reply:
I tried but changed back, because the progress indicator is shared by both pub and sub. The sub does not need thread id since it only has a main thread and a listener thread, while the pub can have a variable number of threads. Making indicator save a thread id as member not only makes no sense for sub, but also the format string is still needed since they are different. One is "(%P|%t) sub %d%% (%d samples received)\n", and the other is "(%P|%t) pub%d  %d%% (%d samples sent)\n". So wrapping the pub thread id in the string seems to be a way to simplify this.